puzzle/plugins/sql/mixins/actions/comment.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+import logging
+
+from puzzle.models.sql import Comment
+
+logger = logging.getLogger(__name__)
+
+
+class CommentActions(object):
+    def comments(self, case_id=None, variant_id=None, username=None):
+        """Return comments for a case or variant.
+
+        Args:
+            case_id (str): id for a related case
+            variant_id (Optional[str]): id for a related variant
+        """
+        logger.debug("Looking for comments")
+        comment_objs = self.query(Comment)
+
+        if case_id:
+            comment_objs = comment_objs.filter_by(case_id=case_id)
+
+        if variant_id:
+            comment_objs = comment_objs.filter_by(variant_id=variant_id)
+        elif case_id:
```

Comment:
yup, but aren't you overwriting the `comment_objs ` variable? So that if `case_id`, then `comment_objs ` will only be the result of `comment_objs.filter_by(variant_id=variant_id)`